pkg/kubectl/cmd/apply_test.go

```diff
@@ -991,6 +998,97 @@ func TestRunApplySetLastApplied(t *testing.T) {
 	cmdutil.BehaviorOnFatal(func(str string, code int) {})
 }
 
+func TestRunApplyDiffLastApplied(t *testing.T) {
+	initTestErrorHandler(t)
+	rcByte := readBytesFromFile(t, filenameApplyDiffRESP)
+	pathRC := "/namespaces/test/replicationcontrollers/" + "test-rc"
+
+	rcChangedByte := readBytesFromFile(t, filenameApplyDiffChangedRESP)
+
+	tests := []struct {
+		name, nameRC, pathRC, filePath, expectedGitDiffOut, expectedDiffOut, expectedErr, output string
+		resp                                                                                     []byte
+	}{
+		{
+			name:               "git-diff objects with no different",
+			filePath:           filenameRC,
+			expectedErr:        "",
+			expectedGitDiffOut: "",
+			expectedDiffOut:    "",
+			resp:               rcByte,
+			output:             "yaml",
+		},
+		{
+			name:               "git-diff after local file changed with json",
+			filePath:           filenameLocalChangedRC,
+			expectedErr:        "",
+			expectedGitDiffOut: "-                \"containerPort\": 80\n+                \"containerPort\": 123\n               }\n",
+			expectedDiffOut:    "<                 \"containerPort\": 80\n>                 \"containerPort\": 123\n",
+			resp:               rcByte,
+			output:             "json",
+		},
+		{
+			name:               "git-diff after annotations changed",
+			filePath:           filenameRC,
+			expectedErr:        "",
+			expectedGitDiffOut: "-        - containerPort: 123\n+        - containerPort: 80\n",
+			expectedDiffOut:    "<         - containerPort: 123\n>         - containerPort: 80\n",
+			resp:               rcChangedByte,
+			output:             "yaml",
+		},
+	}
+	for _, test := range tests {
+		f, tf, _, _ := cmdtesting.NewAPIFactory()
+		tf.Printer = &testPrinter{}
+		tf.UnstructuredClient = &fake.RESTClient{
+			APIRegistry:          api.Registry,
+			NegotiatedSerializer: unstructuredSerializer,
+			Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+				switch p, m := req.URL.Path, req.Method; {
+				case p == pathRC && m == "GET":
```

Comment:
because I follow the previous coding style, seehttps://github.com/shiywang/kubernetes/blob/502bfdf94410f597a86a1471a945b1520979dac6/pkg/kubectl/cmd/create_secret_test.go#L43-L43, https://github.com/shiywang/kubernetes/blob/a82c4481f30e5877edd614f6dc7a323b5e4caf67/pkg/kubectl/cmd/create_test.go#L52-L52 and https://github.com/shiywang/kubernetes/blob/502bfdf94410f597a86a1471a945b1520979dac6/pkg/kubectl/cmd/create_namespace_test.go#L38-L38 ...